### PR TITLE
FAI-463: Weighted random sampler

### PR DESCRIPTION
**Pull request by RobGeada**
[FAI-463](https://issues.redhat.com/browse/FAI-463). The SHAP kernel explainer needs to perform weighted sampling with replacement from a list of items. RandomChoice provides a random sampler that provides these features.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by RobGeada** [comment]
jenkins retest this please

File: explainability/explainability-core/src/test/java/org/kie/kogito/explainability/utils/RandomChoiceTest.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import java.util.List;
+import java.util.Random;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RandomChoiceTest {
+    List<String> obj = List.of("a", "b", "c", "d", "e");
+    Random rn = new Random();
```
Comment (ruivieira) [inline explainability/explainability-core/src/test/java/org/kie/kogito/explainability/utils/RandomChoiceTest.java]: Perhaps parameterising the random instance (with a few different seeds, [example](https://github.com/kiegroup/kogito-apps/blob/4e1b2c8f83e6ae0fc35e2cfa15fcb03a0ffcf80f/explainability/explainability-core/src/test/java/org/kie/kogito/explainability/local/counterfactual/CounterfactualExplainerTest.java#L103))?

**Review by ruivieira — CHANGES_REQUESTED** [review]
@RobGeada lgtm (comments inlined).

(note: I'm not sure what the `TreeMap` overhead is, compared to directly calculating the weights CDF, but that's not an issue for me).

**Review by danielezonca — APPROVED** [review]

File: explainability/explainability-core/src/test/java/org/kie/kogito/explainability/utils/RandomChoiceTest.java (hunk 1 above)
Comment (RobGeada) [reply explainability/explainability-core/src/test/java/org/kie/kogito/explainability/utils/RandomChoiceTest.java]: Added!

**Review by ruivieira — APPROVED** [review]
lgtm, thanks!

**Pull request merged**